### DISCUSSION TITLE
Problems installing multiple plugins

### DISCUSSION THREAD

**Discussion by redgospel** [netbox-community/netbox#17367] [General]
Dear, when I install several plugins and configure them in the configuration.py file:

Plugins = [
"netbox_inventory",
"ipfabric_netbox",
"netbox_bgp",
"netbox_floorplan",
"netbox_dns",
"netbox_qrcode",
"netbox_reorder_rack",
"netbox_topology_views",
"netbox_branching",
"slurpit_netbox",
"nb_risk",
"netbox_acls",
"netbox_config_diff",
"netbox_contract",
"netbox_interface_synchronization",
"netbox_otp_plugin",
"netbox_secrets",
"validity",
]

the netbox server crashes, it only allows me to enable the 1st plugin. the process i followed to install was: 
1. install all the plugins 
2. then execute the commands
        python3 /opt/netbox/netbox/manage.py migrate
        python3 /opt/netbox/netbox/manage.py collectstatic
        sudo systemctl restart netbox netbox-rq
3. I tested and it stopped working, except if I only enable the 1st plugin it works..

what steps should I follow or where should the problem be?.

**Comment by candlerb** [comment]
"Several" plugins - an understatement! :-)

You didn't mention your Netbox version, but many plugins have not been updated for 4.0, or the recently-released 4.1.  Nor did you show what error you get when your Netbox crashes. If it's a Netbox exception, then setting `DEBUG=True` in configuration.py will give you a backtrace, which may identify the offending plugin.

Apart from that, I suggest you enable the plugins one by one, to check they all work individually. (If any is broken, leave it disabled).

Then you can try to see if there are combinations of plugins which are interfering with each other, by a binary chop. That is: enable half of them; if they're fine then enable half of the remainder, and so on.

**Comment by gadler01** [comment]
I'm having the same issue.  I don't have as many plugins :-) and have tried different formats for the PLUGINS = [] list.
 
PLUGINS = [
    "nextbox_ui_plugin",
#    "netbox_qrcode",
    "netbox_floorplan",
    "netbox_reorder_rack",
]

I have tried ' single and " double quotes without success.  

From configuration.py:

/opt/netbox$ python3 /opt/netbox/netbox/manage.py migrate
Traceback (most recent call last):
  File "/opt/netbox-4.1.1/venv/lib/python3.12/site-packages/django/apps/registry.py", line 158, in get_app_config
    return self.app_configs[app_label]
           ~~~~~~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'netbox_floorplan'

During handling of the above exception, another exception occurred:

The error simply shifts depending on what is commented out leading me to believe there is something in the way I have the list formatted.

**Comment by gadler01** [comment]
PLUGINS = [
    "nextbox_ui_plugin",
"#    "netbox_qrcode","
    "netbox_floorplan",
    "netbox_reorder_rack",
]

**Comment by candlerb** [comment]
Use three backticks (` ``` `) on a line of their own, before and after your post, to stop the funky markdown formatting.

> KeyError: 'netbox_floorplan'

It seems to me that either:
1. You've not installed the plugin into your virtualenv (put it in `/opt/netbox/local_requirements.txt` before running `./upgrade.sh`); or
2. You're using the wrong name for the plugin module. Often the pip name of the package is not identical to the module name.

For example, in `local_requirements.txt` I have

```
netbox-topology-views
```

but in configuration.py I have:

```
PLUGINS = [
            'netbox_topology_views',
]
```

(note dash versus underscore)

Assuming that the plugin you're talking about is https://github.com/netbox-community/netbox-floorplan-plugin then it has a similar issue, and the installation instructions are unclear: they're suggesting you do "pip install ..." without reminding you to activate the virtualenv first, but also `./upgrade.sh` destroys and recreates the virtualenv anyway, so you'll lose any manually installed packages. That's why `local_requirements.txt` is the safer way to do it.

**Reply by gadler01** [reply]
Thank you for your response! and for the (```)advice.

All the plugins have been installed with the venv active. cd 

Local requirements is:

cat local_requirements.txt:
nextbox-ui-plugin
netbox-qrcode
netbox-floorplan-plugin
netbox-reorder-rack

Dependencies and installation appear to be satisfied:

/opt/netbox$ pip install netbox-qrcode
Requirement already satisfied: netbox-qrcode in ./venv/lib/python3.12/site-packages (0.0.13)
Requirement already satisfied: qrcode in ./venv/lib/python3.12/site-packages (from netbox-qrcode) (7.4.2)
Requirement already satisfied: Pillow in ./venv/lib/python3.12/site-packages (from netbox-qrcode) (10.4.0)
Requirement already satisfied: typing-extensions in ./venv/lib/python3.12/site-packages (from qrcode->netbox-qrcode) (4.12.2)
Requirement already satisfied: pypng in ./venv/lib/python3.12/site-packages (from qrcode->netbox-qrcode) (0.20220715.0)

(venv):/opt/netbox$ pip install nextbox-ui-plugin
Requirement already satisfied: nextbox-ui-plugin in ./venv/lib/python3.12/site-packages (0.15.0)
(venv):/opt/netbox$ pip install netbox-floorplan-plugin
Requirement already satisfied: netbox-floorplan-plugin in ./venv/lib/python3.12/site-packages (0.4.0)
(venv) :/opt/netbox$ pip install netbox-reorder-rack
Requirement already satisfied: netbox-reorder-rack in ./venv/lib/python3.12/site-packages (1.1.1)

So, to me it appears that the handling of the list in configuration.py is the culprit.

**Reply by candlerb** [reply]
> So, to me it appears that the handling of the list in configuration.py is the culprit.

Quite possibly. Perhaps you should comment them all out, and uncomment them one by one until you find the culprit.

**Reply by gadler01** [reply]
Brian, Thank you for your gracious assistance!  I have commented them out one by one.  It appears the netbox_fl__initoorplan plugin is set with a max version of 4.0.10 and I'm running 4.1.1.  Interesting outcome as I can run the migration after updating the __init__.py but it reverts when I run upgrade.sh.


Thank you!
